packages/valory/skills/abstract_round_abci/behaviour_utils.py

```diff
@@ -1706,31 +1708,33 @@ def reset_tendermint_with_wait(
                 response = json.loads(result.body.decode())
                 if response.get("status"):
                     self.context.logger.info(response.get("message"))
-                    self.context.logger.info(
-                        "Resetting tendermint node successful! Resetting local blockchain."
-                    )
-                    self.context.state.round_sequence.reset_blockchain(
-                        response.get("is_replay", False)
-                    )
+                    self.context.logger.info("Resetting tendermint node successful!")
+                    is_replay = response.get("is_replay", False)
+                    if is_replay:
+                        # in case of replay, the local blockchain should be set up differently.
+                        self.context.state.round_sequence.reset_blockchain(
+                            is_replay=is_replay, is_init=True
+                        )
```

Comment:
Not clear from the code what `is_replay` refers to and what `is_init` actually does.

Reply:
@angrybayblade please provide your insights

Reply:
https://github.com/valory-xyz/open-autonomy/blob/feb545481f07a689b3d636c8d04c7b3faf8684f9/packages/valory/skills/abstract_round_abci/base.py#L2643-L2646

When resetting the blockchain we need to set the block construction phase to `WAITING_FOR_BEGIN_BLOCK` for a reason I cannot remember entirely. But I think the agent was getting stuck after performing a reset in a replay that's why I added this step to avoid that

Reply:
We still do, on line 1700. Resetting for a second time doesn't change anything, as the blockchain is empty at this point. It simply changes this `_block_construction_phase` param.